src/dhcp/aca_dhcp_server.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_dhcp_server.h"
+#include "aca_log.h"
+#include "goalstateprovisioner.grpc.pb.h"
+#include <errno.h>
+#include <arpa/inet.h>
+
+using namespace std;
+using namespace alcorcontroller;
+using namespace aca_dhcp_programming_if;
+
+namespace aca_dhcp_server
+{
+ACA_Dhcp_Server::ACA_Dhcp_Server()
+{
+	try{
+		_dhcp_db = new std::map<string, dhcp_entry_data, dhcp_entry_comp>;
+	} catch (const bad_alloc &e){
+		return;
+	}
+
+	_dhcp_entry_thresh = 0x10000; //10K
```

Comment:
correct me if I am wrong, wasn't that 65536? 

One large VPC can have up to 65536 ports, do we want to limit our DHCP server with this number?